### PR TITLE
Finish some TODOs in `core/users.rs` module

### PR DESCRIPTION
**Pull request by zolagonano**

### CODE REVIEW

File: src/core/user.rs [hunk 1]
```diff
@@ -309,15 +313,22 @@ impl User {
...
-    pub fn get_new_site_data(&mut self) -> SiteData {
-        //TODO: Add option to get a new site without master seed
-        let (site_privkey, site_address, bip32_idx) =
-            self.get_site_keypair_from_seed(&self.master_seed, None);
-
-        if let Some(_site_address) = self.sites.get(&site_address.to_string()) {
-            // TODO: do the whole process again instead of warning
-            info!("Random error: site exist!");
-        }
+    pub fn get_new_site_data(&mut self, without_seed: bool) -> SiteData {
```
Comment (canewsin) [inline src/core/user.rs:316]: rename param to with_seed

**Review by canewsin — CHANGES_REQUESTED** [review]
.

**Comment by zolagonano** [comment]
@canewsin Renamed it!

File: src/core/user.rs [hunk 2]
```diff
@@ -309,15 +317,22 @@ impl User {
...
+                let keypair = self.get_site_keypair_from_seed(&self.master_seed, None);
+                if self.sites.get(&keypair.1).is_none() {
+                    break keypair;
+                }
+            } else {
+                let keypair = self.generate_site_keypair();
+                if self.sites.get(&keypair.1).is_none() {
+                    break keypair;
+                }
+            }
```
Comment (canewsin) [inline src/core/user.rs:328]: prefer 
```rust
let keypair = if with_seed {
    self.get_site_keypair_from_seed(&self.master_seed, None)
} else {
    self.generate_site_keypair()
};
if self.sites.get(&keypair.1).is_none() {
    break keypair;
}
```

**Comment by zolagonano** [comment]
@canewsin done!

**Pull request merged**